Dockerfile

```diff
@@ -1,4 +1,30 @@
+#-----------------------------------------------------#
+#                     System Base                     #
+#-----------------------------------------------------#
 FROM openjdk:8u171-alpine3.7
 RUN apk --no-cache add curl
-COPY target/oncostore*.jar oncostore.jar
-CMD java ${JAVA_OPTS} -jar oncostore.jar
\ No newline at end of file
+
+#-----------------------------------------------------#
+#                Install Variantstore                 #
+#-----------------------------------------------------#
+# Define install directory
+ENV variantstore=/root/variantstore
+RUN mkdir ${variantstore}
+
+# Get required files for setup
+COPY ./src ${variantstore}/src
+COPY ./pom.xml ${variantstore}/pom.xml
+COPY ./micronaut-cli.yml ${variantstore}/micronaut-cli.yml
+COPY ./.mvn ${variantstore}/.mvn
+COPY ./mvnw ${variantstore}/mvnw
+COPY ./mvnw.cmd ${variantstore}/mvnw.cmd
+
+# Compile Variantstore
+RUN cd ${variantstore} && \
+    ./mvnw clean compile
+
+#-----------------------------------------------------#
+#                   Run Variantstore                  #
+#-----------------------------------------------------#
+WORKDIR ${variantstore}
+CMD ./mvnw exec:exec
```

Comment:
Are there advantages/disadvantages in comparison to calling the `.jar` file?

Reply:
Good question.
Sadly, I do not have a large expertise in maven projects.

I would guess that packing it in a jar give you more control on the java environment instead of relying on maven.

Packing a jar & execute it allow maybe more options: I'm thinking of the possibility of adding a volume in which you can access the created packed jar from outside of the container. Maybe, even shutdown the container, replace the jar and restart again for using a newer version.

However, I would assume the separation of the variantstore apart from the db, allow and encourage the option of compiling everything again in a new container which would be cleaner, I guess.

Another option would be to use 2 containers.
1x for purely compiling & packaging the jar
1x for application which just mounts the jar and runs it

This setup would allow self compiling from source but also accessing & replacing the jar if desired.

#open-for-discussion

Reply:
I would suggest to just use `mvn clean package` and the resulting `.jar` file for now if there are no strong arguments against it. We can still change it later if needed.

Reply:
Linked to #47